packages/inferno-animation/src/animations.ts

```diff
@@ -25,66 +26,91 @@ function getAnimationClass(animationProp: AnimationClass | string | undefined |
 }
 
 export function componentDidAppear(dom: HTMLElement, props) {
-  const { start, end, active } = getAnimationClass(props.animation, '-enter');
+  // Get dimensions and unpack class names
+  const cls = getAnimationClass(props.animation, '-enter');
 
-  // 1. Get height and set start of animation
-  const { width, height } = getDimensions(dom);
-  addClassName(dom, start);
-  forceReflow();
-
-  // 2. Activate transition
-  addClassName(dom, active);
-
-  // 3. Set an animation listener, code at end
-  // Needs to be done after activating so timeout is calculated correctly
-  registerTransitionListener(
-    [dom],
-    function () {
-      // *** Cleanup ***
-      // 5. Remove the element
-      clearDimensions(dom);
-      removeClassName(dom, active);
-      removeClassName(dom, end);
+  // Moved measuring to pre_initialize. It causes a reflow for each component beacuse of the setDisplay of previous component.
+  const dimensions = {};
+  const display = setDisplay(dom, 'none');
+  queueAnimation((phase) => _didAppear(phase, dom, cls, dimensions, display));
+}
 
-      // 6. Call callback to allow stuff to happen
-      // Not currently used but this is where one could
-      // add a call to something like this.didAppearDone
-    }
-  );
+function _didAppear (phase: AnimationPhase, dom: HTMLElement, cls: AnimationClass, dimensions, display: string) {
+  switch (phase) {
```

Comment:
this is pretty smart actually, I like it :P

Reply:
Thank you! It has been a good week :)